File: Frontend/src/lib/services/assets.ts
import { Asset } from "@/interfaces/entities/asset";
import { Result } from "@/types/result";
import { invoke } from "@tauri-apps/api/core";

export async function select_assets(env: string): Promise<Result<Asset[]>> {
    try {
        let data = await invoke<Asset[]>("select_assets", {
            env,
        });

        return { ok: true, data };
    } catch (error) {
        return { ok: false, error };
    }
}

File: Frontend/src-tauri/src/command/assets.rs
use crate::{
    core::select_assets_core,
    models::{entities::assets::Model, structs::AssetRequest},
};

#[tauri::command]
pub async fn select_assets(env: String, asset: Option<AssetRequest>) -> Result<Vec<Model>, String> {
    select_assets_core(env, asset).await
}
